File: routes/POST/newProfile.py
from classes import profiles
from flask import make_response, jsonify

def create(profileData):
    try:
        profile = profiles.Profile(name=profileData["name"],photo=profileData["photo"],userId=profileData["userId"])
        createProfile = profile.create()
        match createProfile["message"]:
            case "The profile has been created":
                resp = make_response(jsonify(createProfile))
                resp.status_code = 201
                return resp
            case "An error has occurred while creating the profile":
                resp = make_response(jsonify(createProfile))
                resp.status_code = 500
                return resp
    except Exception as e:
        if e.args[0]=="name"or e.args[0]=="photo"or e.args[0]=="userId":
            resp = make_response(jsonify({"message":"Please, add the following fields: {}".format(e.args[0])}))
            resp.status_code = 400
            return resp

        resp = make_response(jsonify({"message":"There was an error", "error":e.args}))
        resp.status_code = 500
        return resp

File: src/routes/POST/newProfile.ts
import Profile from "../../classes/profiles";
import type { Response, Request } from "express";

export default async function newProfile(req: Request, res: Response){
  try {
    const userId = req.params.userId;
    const { name, photo } = req.body;
    const newProfile =await new Profile("", name, photo, userId).new();

    switch (newProfile.message) {
      case "User not found":
        res.statusCode = 404;
        res.json(newProfile);
        break;
      case "success":
        res.statusCode = 201;
        res.json(newProfile);
        break;
      case "An error has occurred while creating profile":
        res.statusCode = 500;
        res.json(newProfile);
    }
  } catch (error: any) {
    res.statusCode = 500;
    res.json({ message: "There was an error", error: error.message });
  }
}
